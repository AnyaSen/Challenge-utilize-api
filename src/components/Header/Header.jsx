import React from 'react';
import Styles from './Header.module.scss';
import { Link } from 'react-scroll';
import { Link as RouterLink } from 'react-router-dom';

import PrimaryButton from '../Buttons/PrimaryButton/PrimaryButton';
import SecondaryButton from '../Buttons/SecondaryButton/SecondaryButton';

export default function Header() {
	return (
		<div className={Styles.HeaderContainer}>
			<h1>Welcome to Elisa Viihde!</h1>

			<h3>In this app you can see a list of live TV programs and channels.</h3>

			<div className={Styles.buttonsContainer}>
				<Link
					to='programs'
					spy={true}
					smooth={true}
					hashSpy={true}
					offset={-114}
					duration={500}>
					<PrimaryButton text='Live Programs' />
				</Link>

				<RouterLink to='/channels'>
					<SecondaryButton text='All Channels' />
				</RouterLink>
			</div>
		</div>
	);
}
