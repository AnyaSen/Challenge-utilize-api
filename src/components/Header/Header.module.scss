@import "../../assets/scss/variables";
@import "../../assets/scss/breakpoints";

.HeaderContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 88vh;
  padding-bottom: 10rem;
  background-color: $color-dark-white;
  text-align: center;

  @include md {
    padding-bottom: 0;
  }

  h1 {
    color: $color-blue;
    padding: $space-2;
  }

  h3 {
    margin: $space-3 0;
    padding: 0 $space-2;
  }

  .buttonsContainer {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;

    &:first-child {
      margin-right: $space-5;

      @include sm {
        margin-right: $space-3;
      }
    }
  }
}
