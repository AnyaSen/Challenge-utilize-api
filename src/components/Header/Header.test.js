import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import Header from './Header';

describe('<Header />', () => {
	afterAll(cleanup);

	const tree = (
		<BrowserRouter>
			<Header />
		</BrowserRouter>
	);

	const { getByText, container } = render(tree);

	test('Should contain the correct text', () => {
		const mainMessage = 'Welcome to Elisa Viihde!';
		const subMessage =
			'In this app you can see a list of live TV programs and channels.';
		const programsLinkMessage = 'Live Programs';
		const channelsLinkMessage = 'All Channels';

		const mainText = getByText(mainMessage);
		const subText = getByText(subMessage);
		const programsLinkText = getByText(programsLinkMessage);
		const channelsLinkText = getByText(channelsLinkMessage);

		expect(mainText).toBeTruthy();
		expect(subText).toBeTruthy();
		expect(programsLinkText).toBeTruthy();
		expect(channelsLinkText).toBeTruthy();

		expect(container.firstChild).toMatchSnapshot();
	});
});
