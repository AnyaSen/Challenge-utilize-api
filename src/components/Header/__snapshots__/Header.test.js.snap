// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should contain the correct text 1`] = `
<div
  class="HeaderContainer"
>
  <h1>
    Welcome to Elisa Viihde!
  </h1>
  <h3>
    In this app you can see a list of live TV programs and channels.
  </h3>
  <div
    class="buttonsContainer"
  >
    <a>
      <button
        class="PrimaryButton"
      >
        Live Programs
      </button>
    </a>
    <a
      href="/channels"
    >
      <button
        class="SecondaryButton"
      >
        All Channels
      </button>
    </a>
  </div>
</div>
`;
